source/slang/ir.cpp

```diff
@@ -4639,7 +4723,8 @@ namespace Slang
     // are known, and specialize the callee based on those
     // known values.
     void specializeGenerics(
-        IRModule*   module)
+        IRModule*   module,
+        CodeGenTarget target)
```

Comment:
I assume that `target` is now relevant because we have to deal with all the complexity of target-specific function selection that the initial cloning step has to, even though this is supposed to be just a generic specialization pass.

This just seems to confirm to be that the phasing here is broken, and we *need* to be doing the copy step (where we grab the best definitions for the target, for all the symbols that are needed) first, and then only apply specialization when that is done (after which we can ignore all the issues around multiple values with the same mangled name).